### ISSUE TITLE
Lossless gzip compression of floating-point fits image data puts incorrect ZQUANTIZ header value

### ISSUE THREAD

**Issue by erykoff** [astropy/astropy#10554]
### Description

Writing floating point image data with lossless gzip compression (specifying `quantize_level=0.0`) sets `ZQUANTIZ='NO_DITHER'` in the compressed HDU header.  This is incompatible with cfitsio and funpack, which expects `ZQUANTIZ='NONE'` for unquantized data.  If you run `funpack` on a file compressed in this way you get garbage data (but no error is raised).  Note that this bug is only triggered when you explicitly set `quantize_level=0.0`, and is not triggered when quantization is automatically turned off when the fits writer cannot compute a good RMS.

### Expected behavior

What I expect is that compressed data written by `astropy.io.fits` is readable by other standard fits readers.

### Actual behavior

What actually happens is that you get garbage data when running `funpack` on an image compressed in this way.

### Steps to Reproduce

The following code assumes that `funpack` is available in the system path.

```python
import astropy.io.fits as fits
import numpy as np
import subprocess
import os
import mmap


arr = np.random.normal(size=(1000, 1000), loc=100.0, scale=10.0)

hdu_list = fits.HDUList()
hdu = fits.CompImageHDU(data=arr, header=fits.Header(),
                        compression_type='GZIP_2', quantize_level=0)
hdu_list.append(hdu)
hdu_list.writeto('test_gzip_compressed_image.fits.fz', overwrite=True)

arr2 = fits.getdata('test_gzip_compressed_image.fits.fz', ext=0)

# This test succeeds because astropy.io.fits can read these files
np.testing.assert_array_almost_equal(arr2, arr)

# Use funpack to uncompress test_gzip_compressed_image.fits.fz
if os.path.isfile("test_gzip_compressed_image.fits"):
    os.unlink("test_gzip_compressed_image.fits")
subprocess.run(["funpack", "test_gzip_compressed_image.fits.fz"])

arr3 = fits.getdata('test_gzip_compressed_image.fits', ext=0)

# This will fail because the funpack uncompress failed (silently)
np.testing.assert_array_almost_equal(arr3, arr)

# Replace the ZQUANTIZ= 'NO_DITHER' header value with NONE
# Do this manually because the Z* header keywords are reserved
with open('test_gzip_compressed_image.fits.fz', "r+b") as f:
    mm = mmap.mmap(f.fileno(), 0)
    loc = mm.find(b"ZQUANTIZ= 'NO_DITHER'")
    if loc >= 0:
        mm.seek(loc)
        mm.write(b"ZQUANTIZ= 'NONE     '")

# Use funpack to uncompress the updated version
if os.path.isfile("test_gzip_compressed_image.fits"):
    os.unlink("test_gzip_compressed_image.fits")
subprocess.run(["funpack", "test_gzip_compressed_image.fits.fz"])

arr3 = fits.getdata('test_gzip_compressed_image.fits', ext=0)

# The following test works, the arrays are equal.
np.testing.assert_array_almost_equal(arr3, arr)
```

### System Details

Linux-2.6.32-754.27.1.el6.x86_64-x86_64-with-redhat-6.10-Santiago
Python 3.7.6 | packaged by conda-forge | (default, Mar 23 2020, 23:03:20) 
[GCC 7.3.0]
Numpy 1.18.4
astropy 4.0.1.post1
Scipy 1.4.1
Matplotlib 3.0.3

**Comment by saimn** [comment]
There is no mention of the `'NONE'` option for `ZQUANTIZ` in the FITS standard. `ZQUANTIZ` is meant to specify the dithering algorithm, or no dithering. For lossless compression, without quantization, it is mentioned that `ZSCALE` and `ZZERO` should not be set, which is was astropy is doing with the file produced in your example. 

![image](https://user-images.githubusercontent.com/311639/87842945-583c4200-c87e-11ea-9d99-0e96f3eabe6d.png)

So for me it is a bug in cfitsio. And it is a bit surprising because this comment in astropy seems to indicate that cfitsio used to treat correctly the absence of `ZSCALE` and `ZZERO` :
https://github.com/astropy/astropy/blob/25ac198cfe0040d34f139ae6d7417c3a3e12768c/astropy/io/fits/hdu/compressed.py#L865-L867

**Comment by saimn** [comment]
Some good news from cfitsio's help desk :
> We have made a change to CFITSIO that will appear in the next version.  For compressed files, if it can't find ZSCALE and ZZERO it will automatically turn quantization off.  That is, it will treat the file exactly as if it had a ZQUANTIZ keyword set to 'NONE' (even if ZQUANTIZ is set to something else such 'NO_DITHER).

And on astropy's side, I will see if it's not too diffcult to support setting ZQUANTIZ to 'NONE', to support older versions of cfitsio.

**Comment by saimn** [comment]
cfitsio 3.49 is out with a fix for this issue (#10663).
I had a look at ZQUANTIZ to 'NONE' but it is not so easy, so I guess we can just ask people having this issue to use a more recent version of cfitsio / funpack.

**Comment by pllim** [comment]
Always nice to have a problem that fixes itself! 🎉
